cloud/db/services/projects.test.ts

```diff
@@ -0,0 +1,902 @@
+            user.id,
+          );
+
+          yield* db.projects.create(
+            { name: "Project 2", organizationId: org2.id },
+            user.id,
+          );
+
+          const projects = yield* db.projects.findAll(user.id);
+          expect(projects).toHaveLength(2);
+          expect(projects.find((p) => p.name === "Project 1")).toBeDefined();
+          expect(projects.find((p) => p.name === "Project 2")).toBeDefined();
+        }),
+      ),
+    );
+
+    it(
+      "should return empty array for user with no projects",
+      withTestDatabase((db) =>
+        Effect.gen(function* () {
+          const user = yield* db.users.create({
+            email: "test@example.com",
+            name: "Test User",
+          });
+
+          const projects = yield* db.projects.findAll(user.id);
+          expect(projects).toHaveLength(0);
+        }),
+      ),
+    );
+  });
+
+  describe("findByOrganization", () => {
+    it(
+      "should retrieve all projects in an organization",
+      withTestDatabase((db) =>
+        Effect.gen(function* () {
+          const user = yield* db.users.create({
+            email: "test@example.com",
+            name: "Test User",
+          });
+
+          const org = yield* db.organizations.create(
+            { name: "Test Organization" },
+            user.id,
+          );
+
+          yield* db.projects.create(
+            { name: "Project A", organizationId: org.id },
+            user.id,
+          );
+
+          yield* db.projects.create(
+            { name: "Project B", organizationId: org.id },
+            user.id,
+          );
+
+          const projects = yield* db.projects.findByOrganization(
+            org.id,
+            user.id,
+          );
+          expect(projects).toHaveLength(2);
+          expect(projects.find((p) => p.name === "Project A")).toBeDefined();
+          expect(projects.find((p) => p.name === "Project B")).toBeDefined();
+        }),
+      ),
+    );
+
+    it(
+      "should return PermissionDeniedError when user is not a member",
+      withTestDatabase((db) =>
+        Effect.gen(function* () {
+          const owner = yield* db.users.create({
+            email: "owner@example.com",
+            name: "Owner",
+          });
+
+          const otherUser = yield* db.users.create({
+            email: "other@example.com",
+            name: "Other User",
+          });
+
+          const org = yield* db.organizations.create(
+            { name: "Test Organization" },
+            owner.id,
+          );
+
+          const result = yield* Effect.either(
+            db.projects.findByOrganization(org.id, otherUser.id),
+          );
+
+          expect(result._tag).toBe("Left");
+          if (result._tag === "Left") {
+            expect(result.left).toBeInstanceOf(PermissionDeniedError);
+            expect(result.left.message).toBe(
+              "You do not have permission to read projects in this organization",
```

Comment:
Seems sketchy that we're using `result._tag` rather than the public interface, eg:

```
const result = yield* Effect.either(
  db.projects.findByOrganization(org.id, otherUser.id),
);

expect(Either.isLeft(result)).toBe(true);
if (Either.isLeft(result)) {
  expect(result.left).toBeInstanceOf(PermissionDeniedError);
  expect(result.left.message).toBe("User is not a member of this organization");
}
```